src/header/header.scss

```diff
@@ -190,181 +136,191 @@
+  @include iui-button-borderless;
   @include iui-button-size(large, borderless);
-
+  @include iui-header-button-reset;
   gap: 0;
   max-width: 25vw;
-  height: 100%;
-  border-radius: 0;
   overflow: hidden;
   padding-right: $iui-s;
 
-  @include iui-header-button-focus;
+  &-name,
+  &-description {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    transition: color $iui-speed-fast ease-out;
+  }
+
+  &-description {
+    height: $iui-baseline * 2;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: height $iui-speed-fast ease-out;
+    }
+    @include themed {
+      opacity: t(iui-opacity-3);
+    }
+  }
 
   > .iui-button-label {
     text-align: left;
     font-size: $iui-font-size;
     overflow: hidden;
 
-    > div {
-      overflow: hidden;
-      text-overflow: ellipsis;
-    }
-
-    .iui-description {
-      height: $iui-baseline * 2;
-      @media (prefers-reduced-motion: no-preference) {
-        transition: all $iui-speed-fast ease;
-      }
-      @include themed {
-        opacity: t(iui-opacity-3);
-      }
+    @media (max-width: 768px) {
+      margin-left: 0;
     }
   }
 
   > .iui-button-icon {
     flex-shrink: 0;
+    box-sizing: content-box;
   }
 
   > * + * {
     margin-left: $iui-sm;
   }
 
   .iui-header-button-icon {
+    display: inline-flex;
     padding: $iui-xs;
+    box-sizing: content-box;
     @include iui-header-button-icon($iui-icons-default, $iui-border-radius);
     @include themed {
       background-color: t(iui-color-background-3);
       fill: t(iui-icons-color);
     }
+
+    @media (max-width: 768px) {
+      display: none;
+    }
+  }
+
+  img.iui-header-button-icon {
+    padding: 0;
+    width: auto;
+    height: $iui-icons-default + ($iui-xs * 2);
   }
 
   &.iui-active {
     @include iui-header-button-active;
   }
 
-  &:disabled {
+  &.iui-disabled,
+  &:disabled,
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+
     .iui-header-button-icon {
       filter: grayscale(1) opacity(0.7);
     }
   }
 
   @at-root {
-    .iui-page-header.iui-slim {
-      > .iui-left {
-        .iui-button {
-          .iui-header-button-icon.iui-header-button-icon {
-            width: $iui-icons-small;
-            height: $iui-icons-small;
-            padding: $iui-xxs;
-          }
+    .iui-slim {
+      .iui-header-breadcrumbs-button {
+        .iui-header-button-icon {
+          width: $iui-icons-small;
+          height: $iui-icons-small;
+          padding: $iui-xxs;
+          margin: 0 $iui-xs;
+        }
+
+        img.iui-header-button-icon {
+          width: $iui-icons-small + ($iui-xxs * 2);
+          height: $iui-icons-small + ($iui-xxs * 2);
+          padding: 0;
         }
 
-        .iui-button-label {
-          .iui-description {
-            @include iui-display-none;
-          }
+        .iui-header-breadcrumbs-button-description {
+          @include iui-display-none;
         }
       }
     }
   }
 }
 
+@mixin iui-header-button {
```

Comment:
This is getting confusing. Why is there a `header-button` and a `header-breadcrumbs-button`?

The example in your description doesn't apply here because those extra buttons are passed by the user.

Reply:
We can support placing 3 types of buttons in the header with unique styling:  a breadcrumb button, a header button, and a regular button (maybe 4 if you count the branding button as well).  Our previous implementation if a user passes a regular button they would get this:
![Screen Shot 2022-02-18 at 4 25 37 PM](https://user-images.githubusercontent.com/849817/154763313-f4cc40d5-653f-4499-891c-837b4b8b9cbb.png)

This `.iui-header-button` modifier ensures that a user passed regular button's styling doesn't get overriden.

Reply:
We cannot apply any modifiers on user's buttons, because we don't know what element they will pass.

Reply:
Ok, this change has been reverted.  All `.iui-button` within the right section will be forced 100% height, but other sections are uneffected (which allows the search button to work in the center section).